tools/UpdateModules.ps1

```diff
@@ -26,34 +26,151 @@ function Create-ModulePsm1
+           }
+           elseif ($mod["RequiredVersion"])
+           {
+               $importedModules += "Import-Module " + $mod["ModuleName"] + " -RequiredVersion " + $mod["RequiredVersion"] + "`r`n"
+           }        
+        }
+     }
+
+     if ($ModuleMetadata.NestedModules -ne $null)
+     {
+         foreach ($dll in $ModuleMetadata.NestedModules)
+         {
+             $importedModules += "Import-Module (Join-Path -Path `$PSScriptRoot -ChildPath " + $dll.Substring(2) + ")`r`n"
+         }
      }
+
      $template = Get-Content -Path $TemplatePath
-     $template = $template -replace "%MODULE-NAME%", $module.Name
+     $template = $template -replace "%MODULE-NAME%", $file.BaseName
      $template = $template -replace "%DATE%", [string](Get-Date)
-     $template = $template -replace "%STRICT-DEPENDENCIES%", $strict
-     $template = $template -replace "%DEPENDENCIES%", $loose
+     $template = $template -replace "%IMPORTED-DEPENDENCIES%", $importedModules
+
+     $contructedCommands = Find-DefaultResourceGroupCmdlets -AddDefaultParameters $AddDefaultParameters -ModuleMetadata $ModuleMetadata -ModulePath $ModulePath
+     $template = $template -replace "%COMMANDS%", $contructedCommands
+     
      Write-Host "Writing psm1 manifest to $templateOutputPath"
      $template | Out-File -FilePath $templateOutputPath -Force
      $file = Get-Item -Path $templateOutputPath
   }
 }
 
+function Find-DefaultResourceGroupCmdlets
+{
+    [CmdletBinding()]
+    param(
+        [bool]$AddDefaultParameters,
+        [Hashtable]$ModuleMetadata,
+        [string]$ModulePath
+    )
+    PROCESS
+    {
+        if ($AddDefaultParameters) 
+        {
+        $nestedModules = $ModuleMetadata.NestedModules
+        $AllCmdlets = @()
+        $nestedModules | ForEach-Object {
+            $dllPath = Join-Path -Path $ModulePath -ChildPath $_
+            $Assembly = [Reflection.Assembly]::LoadFrom($dllPath)
+            $dllCmdlets = $Assembly.GetTypes() | Where-Object {$_.CustomAttributes.AttributeType.Name -contains "CmdletAttribute"}
+            $AllCmdlets += $dllCmdlets
+        }
+        
+        $FilteredCommands = $AllCmdlets | Where-Object {Test-RequiredParameter -Cmdlet $_}
+    
+        if ($FilteredCommands.Length -eq 0) {
+            $contructedCommands = "@()"
+        }
+        else {
+            $contructedCommands = "@("
+            $FilteredCommands | ForEach-Object {
+                $contructedCommands += "'" + $_.GetCustomAttributes("System.Management.Automation.CmdletAttribute").VerbName + "-" + $_.GetCustomAttributes("System.Management.Automation.CmdletAttribute").NounName + ":ResourceGroupName" + "',"
+            }
+            $contructedCommands = $contructedCommands -replace ".$",")"
+        }
+    
+        return $contructedCommands
+        }
+
+        else {
+        return "@()"
+        }
+    }
+}
+
+function Test-RequiredParameter
```

Comment:
Perhaps this should be ```Test-CmdletRequiredParameter```  and the parameter name should likely be a parameter.

Reply:
Done